### PR TITLE
media-gfx/netgen: new package, add 6.2.2204

### PR DESCRIPTION
**Pull request by waebbl**
The media-gfx/freecad package has support for meshing using Netgen. 
Talked about in https://github.com/gentoo/sci/pull/1180#issuecomment-1315854894

Signed-off-by: Bernd Waibel <waebbl-gentoo@posteo.net>

### CODE REVIEW

**Comment by waebbl** [comment]
@AndrewAmmerlaan Here's the netgen package for review like we talked on ::sci and IRC.

File: media-gfx/netgen/netgen-6.2.2204.ebuild
Comment (AndrewAmmerlaan) [inline media-gfx/netgen/netgen-6.2.2204.ebuild:6]: Is it compatible with python3_11?

File: media-gfx/netgen/netgen-6.2.2204.ebuild
Comment (AndrewAmmerlaan) [inline media-gfx/netgen/netgen-6.2.2204.ebuild:144]: Are we legally allowed to distribute and mirror this icon file? Sometimes images or logos are distributed under different licenses or are subject to trademarks.

File: media-gfx/netgen/netgen-6.2.2204.ebuild
Comment (waebbl) [reply media-gfx/netgen/netgen-6.2.2204.ebuild:6]: IIRC during the update to this version on ::sci, I tested it and it wasn't.
I can check again, if you insist, else I would leave this to a future update.

File: media-gfx/netgen/netgen-6.2.2204.ebuild
Comment (waebbl) [reply media-gfx/netgen/netgen-6.2.2204.ebuild:144]: Don't know. Need to check this.

File: media-gfx/netgen/netgen-6.2.2204.ebuild
Comment (AndrewAmmerlaan) [reply media-gfx/netgen/netgen-6.2.2204.ebuild:6]: > IIRC during the update to this version on ::sci, I tested it and it wasn't. I can check again, if you insist, else I would leave this to a future update.

Ah yes, you are right, I forgot about that. Lets leave it as is for now and re-check when there is a new release. Python3.11 is preliminary scheduled to become the default around June so we still have plenty of time.

File: media-gfx/netgen/netgen-6.2.2204.ebuild
Comment (waebbl) [reply media-gfx/netgen/netgen-6.2.2204.ebuild:144]: They have OSX and Windows icon files on their repo, but I haven't found one usable for Unix systems.
The file was added back in 2008 according to `git blame`. IMO we **shouldn't** use it without a go from upstream.
I asked upstream at https://github.com/NGSolve/netgen/issues/147.

Could we use the `netgen.ico` for Windows or the `netgen.icns` for OSX? Does freedesktop.org allow for and recognize such files?

Comment (AndrewAmmerlaan) [reply media-gfx/netgen/netgen-6.2.2204.ebuild:144]: > Could we use the `netgen.ico` for Windows or the `netgen.icns` for OSX? Does freedesktop.org allow for and recognize such files?

We can't use the `.ico` directly but we can use the `convert` command from `virtual/imagemagick-tools` to "unpack" the `.ico` file into a bunch of differently sized `.png` files and `doicon -s ..` those. See here for an example: https://github.com/gentoo/guru/blob/master/games-action/technic-launcher/technic-launcher-4.625.ebuild

There is also `media-gfx/icoutils` but I have never used it and it is probably better to use imagemagick/graphicsmagick since it is likely already installed on the users system.

Comment (waebbl) [reply media-gfx/netgen/netgen-6.2.2204.ebuild:144]: > We can't use the `.ico` directly but we can use the `convert` command from `virtual/imagemagick-tools` to "unpack" the `.ico` file into a bunch of differently sized `.png` files and `doicon -s ..` those. See here for an example: https://github.com/gentoo/guru/blob/master/games-action/technic-launcher/technic-launcher-4.625.ebuild

I will test whether we can use this.

Comment (waebbl) [reply media-gfx/netgen/netgen-6.2.2204.ebuild]: The `.ico` contains 2 usable sizes, 16 and 32. Also droppend the `.desktop` file and generated one using `make_desktop_entry`.

File: media-gfx/netgen/netgen-6.2.2204.ebuild
Comment (AndrewAmmerlaan) [inline media-gfx/netgen/netgen-6.2.2204.ebuild:60]: This dependency we can we can put behind a `gui? ( ... )`. Since we only call the convert command if we build the gui.

File: media-gfx/netgen/metadata.xml
Comment (AndrewAmmerlaan) [inline media-gfx/netgen/metadata.xml:15]: Since this is an import from the ::science overlay we can list the science project as the proxy:
```suggestion
	<maintainer type="project" proxied="proxy">
		<email>sci-mathematics@gentoo.org</email>
		<name>Gentoo Mathematics Project</name>
	</maintainer>
	<maintainer type="person" proxied="yes">
		<email>waebbl-gentoo@posteo.net</email>
		<name>Bernd Waibel</name>
	</maintainer>
```

**Review by AndrewAmmerlaan — APPROVED** [review]
Tested and LGTM :+1: . Just some minor comments and then I think this is good to go.

File: media-gfx/netgen/netgen-6.2.2204.ebuild
Comment (waebbl) [reply media-gfx/netgen/netgen-6.2.2204.ebuild:60]: good catch, I overlooked that

File: media-gfx/netgen/metadata.xml
Comment (waebbl) [reply media-gfx/netgen/metadata.xml:15]: same for the proxy maintainers project, I guess

**Review by AndrewAmmerlaan — APPROVED** [review]

File: media-gfx/netgen/metadata.xml
Comment (AndrewAmmerlaan) [reply media-gfx/netgen/metadata.xml]: > same for the proxy maintainers project, I guess

Yes, but if we list proxy-maint as maintainer as well than we need ACK from @gentoo/proxy-maint

Comment (waebbl) [reply media-gfx/netgen/metadata.xml]: Doesn't proxy-maint have to be added any time, because I'm a proxy-maintainer?

Comment (AndrewAmmerlaan) [reply media-gfx/netgen/metadata.xml]: > Doesn't proxy-maint have to be added any time, because I'm a proxy-maintainer?

Not necessarily, there needs to be a proxy, but the proxy does not have to be the proxy-maint project. Any developer/project can be a proxy.

Comment (waebbl) [reply media-gfx/netgen/metadata.xml]: Oh I see, didn't know this. I always assumed, for a proxy maintainer, proxy-maint does always need to be listed as a proxy. Then I will drop the proxy-maint, for ease of use. If someone later expects it to be there, it can always get added again.

**Pull request closed without merging**
